modules/gapi/misc/python/shadow_gapi.hpp

```diff
@@ -3,11 +3,10 @@
 
 namespace cv
 {
-   struct GAPI_EXPORTS_W_SIMPLE GCompileArg { };
-
-   GAPI_EXPORTS_W GCompileArgs compile_args(gapi::GKernelPackage pkg);
-   GAPI_EXPORTS_W GCompileArgs compile_args(gapi::GNetPackage pkg);
-   GAPI_EXPORTS_W GCompileArgs compile_args(gapi::GKernelPackage kernels, gapi::GNetPackage nets);
+   struct GAPI_EXPORTS_W_SIMPLE GCompileArg {
+       GAPI_WRAP GCompileArg(gapi::GKernelPackage pkg);
+       GAPI_WRAP GCompileArg(gapi::GNetPackage pkg);
+   };
```

Comment:
while I like the idea, this API is not quite equivalent. what if, as a user, I have my own `gapi::GMyOwnPackage` - how can I use `compile_args` machinery to wrap this external-to-G-API package?
it seems remotely possible with the old API (just add a new overload of `compile_args`) - albeit, I'm not sure if python has anything like function overload resolution/adl (wondering how this old API works) - if this is out of scope for this (e.g. G-API only supports packages that it provides itself via `compile_args` and users have to do something else), then feel free to ignore this.

Reply:
Wow, glad to see u :) 

As for `gapi::GMyOwnPackage` what do u mean ? Do u want to use your own compile argument ? 
From python user can't define new compile argument.

Reply:
> Wow, glad to see u :)

Just occasionally looking at things here, nothing too serious 😉 

> As for gapi::GMyOwnPackage what do u mean ? Do u want to use your own compile argument ?

Yes, imagine such a use case (maybe someone wants their own package - it is the thing that provides the kernels/etc, right? can't remember anymore).

> From python user can't define new compile argument.

Well, it should be doable from C++ and then C++ gets magically converted into python - I guess, ideally, G-API should be convenient to use from Python as well, so if someone supplies their own package, it could be used from python API as well.
However, this is a purely artificial use case. I wrote this comment in the meaning of:
you change the API and this (might) limit the usability - e.g. the ability to use custom packages in python. if this was never a concern, then this change is perfectly fine.

Reply:
(glad to see you as well!)

Reply:
>>> Yes, imagine such a use case (maybe someone wants their own package - it is the thing that provides the kernels/etc, right? can't remember anymore).

Again, what's the purpose of the `gapi::GMyOwnPackage` ? First of all it won't be recognized inside G-API, right ? Secondly if the user want to pass his own set of kernels it can be easily done by using `cv.gapi.kernels(....)` which returns `gapi::GKernelPackage` 

I mean that user defined compile arguments are useless because there is no support inside G-API to handle them.

Reply:
ah, I get it now. now objections then.